client/my-sites/plans-v2/product-card/index.tsx

```diff
@@ -0,0 +1,184 @@
+import { isProductsListFetching } from 'state/products-list/selectors/is-products-list-fetching';
+import { getProductCost } from 'state/products-list/selectors/get-product-cost';
+import { getSitePurchases } from 'state/purchases/selectors';
+import getSitePlan from 'state/sites/selectors/get-site-plan';
+import getSiteProducts from 'state/sites/selectors/get-site-products';
+import { useLocalizedMoment } from 'components/localized-moment';
+import JetpackPlanCard from 'components/jetpack/card/jetpack-plan-card';
+import JetpackBundleCard from 'components/jetpack/card/jetpack-bundle-card';
+import JetpackProductCard from 'components/jetpack/card/jetpack-product-card';
+import JetpackProductCardUpgradeNudge from 'components/jetpack/card/jetpack-product-card/upgrade-nudge';
+import { TERM_MONTHLY } from 'lib/plans/constants';
+import { isCloseToExpiration } from 'lib/purchases';
+
+/**
+ * Type dependencies
+ */
+import type { PurchaseCallback, SelectorProduct } from '../types';
+
+const itemToCard = ( { type }: SelectorProduct ) => {
+	switch ( type ) {
+		case ITEM_TYPE_PLAN:
+			return JetpackPlanCard;
+		case ITEM_TYPE_BUNDLE:
+			return JetpackBundleCard;
+		case ITEM_TYPE_PRODUCT:
+		default:
+			return JetpackProductCard;
+	}
+};
+
+interface UpgradeNudgeProps {
+	item: SelectorProduct;
+	currencyCode: string;
+	onClick: PurchaseCallback;
+}
+
+const UpgradeNudgeWrapper = ( { item, currencyCode, onClick }: UpgradeNudgeProps ) => {
+	const upgradeToProductSlug =
+		getRealtimeFromDaily( item.costProductSlug || item.productSlug ) || '';
+	const selectorProductToUpgrade = slugToSelectorProduct( upgradeToProductSlug );
+
+	const isFetchingPrices = useSelector( ( state ) => isProductsListFetching( state ) );
+
+	const itemCost = useSelector( ( state ) => getProductCost( state, upgradeToProductSlug ) );
+	const monthlyItemCost = useSelector( ( state ) =>
+		getProductCost( state, selectorProductToUpgrade?.monthlyProductSlug || '' )
+	);
+
+	if ( ! isUpgradeable( item.productSlug ) || isFetchingPrices || ! selectorProductToUpgrade ) {
+		return null;
+	}
+
+	let originalPrice = 0;
+	let discountedPrice = undefined;
+	if ( ! isFetchingPrices && itemCost ) {
+		originalPrice = itemCost;
+		if ( monthlyItemCost && selectorProductToUpgrade.term !== TERM_MONTHLY ) {
+			originalPrice = monthlyItemCost * 12;
+			discountedPrice = itemCost;
+		}
+	}
+
+	return (
+		<JetpackProductCardUpgradeNudge
+			billingTimeFrame={ durationToText( selectorProductToUpgrade.term ) }
+			currencyCode={ currencyCode }
+			discountedPrice={ discountedPrice }
+			originalPrice={ originalPrice }
+			onUpgradeClick={ () => onClick( selectorProductToUpgrade ) }
+			selectorProduct={ selectorProductToUpgrade }
+		/>
+	);
+};
+
+interface ProductCardProps {
+	item: SelectorProduct;
+	onClick: PurchaseCallback;
+	siteId: number | null;
+	currencyCode: string;
+	className?: string;
+}
+
+const ProductCardWrapper = ( {
```

Comment:
This file and component is everything 👍 . I love to have just one interface to work with a bundle/plan/product.